src/components/WalletModal/index.tsx

```diff
@@ -126,101 +133,118 @@ export default function WalletModal({
   confirmedTransactions: string[] // hashes of confirmed
   ENSName?: string
 }) {
-  // important that these are destructed from the account-specific web3-react context
-  const { account, connector, activate, error } = useWeb3React()
+  const dispatch = useAppDispatch()
+  const { connector } = useWeb3React()
+  const isActiveMap: Partial<Record<Wallet, boolean>> = {
```

Comment:
Why is this `Partial`? Should it be all `ModalWallet`s?

Reply:
This should be memoized, or else the object is re-created every render cycle. See the comment starting with "Tighten up all the logic" for suggested logic to build this map.